src/legacy/ui/public/kfetch/kfetch.ts

```diff
@@ -48,32 +47,36 @@ const interceptors: Interceptor[] = [];
 export const resetInterceptors = () => (interceptors.length = 0);
 export const addInterceptor = (interceptor: Interceptor) => interceptors.push(interceptor);
 
-export async function kfetch(
-  options: KFetchOptions,
-  { prependBasePath = true }: KFetchKibanaOptions = {}
-) {
-  const combinedOptions = withDefaultOptions(options);
-  const promise = requestInterceptors(combinedOptions).then(
-    ({ pathname, query, ...restOptions }) => {
-      const fullUrl = url.format({
-        pathname: prependBasePath ? chrome.addBasePath(pathname) : pathname,
-        query,
-      });
-
-      return window.fetch(fullUrl, restOptions).then(async res => {
-        if (!res.ok) {
-          throw new KFetchError(res, await getBodyAsJson(res));
-        }
-        const contentType = res.headers.get('content-type');
-        if (contentType && contentType.split(';')[0] === 'application/ndjson') {
-          return await getBodyAsBlob(res);
-        }
-        return await getBodyAsJson(res);
-      });
-    }
-  );
+export function createKfetch(http: HttpSetup) {
+  return function kfetch(
+    options: KFetchOptions,
+    { prependBasePath = true }: KFetchKibanaOptions = {}
+  ) {
+    const controller = new AbortController();
+    const promise = new Promise((resolve, reject) => {
+      responseInterceptors(
+        requestInterceptors(withDefaultOptions(options))
+          .then(({ pathname, query, ...restOptions }) =>
+            http.fetch(pathname, {
+              ...restOptions,
+              signal: controller.signal,
+              query,
+              prependBasePath,
+            })
+          )
+          .catch(err => {
+            throw new KFetchError(err.response || { statusText: err.message }, err.body);
```

Comment:
cannot find a test for this. did I miss something?

Reply:
There used to be a test for this (I believe), but it's functionality should be caught by the http service tests.